types/stylelint/index.d.ts

```diff
@@ -67,7 +67,12 @@ declare module 'stylelint' {
 	}[keyof T];
 	export type DisablePropertyName = PropertyNamesOfType<StylelintConfig, DisableSettings>;
 
-	export type StylelintCosmiconfigResult = (CosmiconfigResult & { config: StylelintConfig }) | null;
+	// This type has the same properties as `CosmiconfigResult` from `cosmiconfig`.
+	export type StylelintCosmiconfigResult = {
+		config: StylelintConfig;
```

Comment:
[note] `CosmiconfigResult.config` is `any` and not type-safe, so I've replaced the use of `CosmiconfigResult`.